### PR TITLE
Improve App linking workflow 

### PR DESCRIPTION
**Pull request by TripleSpeeder**
Main changes:
- Decouple Sponsoring/App linking logic from UI
- Track App linking state/progress in Redux Store
- Move logic to AppThunks
- Split sponsoring and app linking logic into several Thunks/functions
- Create a UI that better shows what is currently happening
- Provide better feedback in error case
- Fix several typing issues

Changes to profile/social media sync:
- only show option if the user meets all requirements to link with the registry apps
- save user preference per social media type instead of globally
- Don't do any app linking in background. Instead use new modal interface to show app linking progress
- Don't track app link status in socialMedia store, query apps store instead

### CODE REVIEW

File: BrightID/src/components/Apps/appThunks.ts [hunk 1]
```diff
@@ -0,0 +1,588 @@
...
+
+    const api = getGlobalNodeApi();
+    dispatch(setAppLinkingStep({ step: app_linking_steps.REFRESHING_APPS }));
+    try {
+      // make sure to have latest appInfo available
+      const apps = await api.getApps();
+      dispatch(setApps(apps));
+    } catch (e) {
+      const msg = e instanceof Error ? e.message : `${e}`;
+      dispatch(setLinkingAppError(`Failed to fetch latest appInfo: ${msg}`));
```
Comment (abramsymons) [inline BrightID/src/components/Apps/appThunks.ts:73]: It seems if such problems happen, users should reset the app to be able to retry linking. yes?

File: BrightID/src/components/Apps/appThunks.ts [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+
+export const linkAppOrContext =
+  (): AppThunk<Promise<void>> => async (dispatch: AppDispatch, getState) => {
+    const applinkingStep = selectApplinkingStep(getState());
+    if (applinkingStep !== app_linking_steps.SPONSOR_SUCCESS) {
+      console.log(
+        `Can't start linkApp when not in SUCCESS state. Current state: ${applinkingStep}`,
+      );
+    }
+    // TODO How to handle below condition?
```
Comment (abramsymons) [inline BrightID/src/components/Apps/appThunks.ts:142]: In the previous version, a recursive call and a sleep were used to continue linking after blind sigs were updated.
It seems that is removed, and this TODO is missed to be implemented.

File: BrightID/src/components/Apps/appThunks.ts [hunk 3]
```diff
@@ -0,0 +1,588 @@
...
+    }
+
+    if (linkSuccess) {
+      // at least one verification successfully linked
+      // TODO If there were errors with other verifications (sigErrors array), how to show in the UI?
+      dispatch(setAppLinkingStep({ step: app_linking_steps.LINK_SUCCESS }));
+
+      if (appInfo.callbackUrl) {
+        const onSuccess = async () => {
+          if (appInfo.callbackUrl) {
```
Comment (abramsymons) [inline BrightID/src/components/Apps/appThunks.ts:551]: This check seems to be repeated.

File: BrightID/src/components/EditProfile/socialMediaThunks.ts [hunk 4]
```diff
@@ -32,7 +31,8 @@ export async function linkSocialMediaApp(appId: string, appUserId: string) {
       Date.now() - signedTimestamp > SOCIAL_MEDIA_SIG_WAIT_TIME
     ) {
       try {
-        linked = await linkAppId(appId, appUserId, true);
+        // TODO
```
Comment (abramsymons) [inline BrightID/src/components/EditProfile/socialMediaThunks.ts:34]: This also seems to be missed to implement.
Linking social media apps do not need sponsoring or user request and do not follow general linking routine.
One of the things seems should not be applied to this linking is this check in this beginning of the `linkAppId`:
`if (applinkingStep !== app_linking_steps.SPONSOR_SUCCESS) {

File: BrightID/src/components/Apps/appThunks.ts (hunk 2 above)
Comment (TripleSpeeder) [reply BrightID/src/components/Apps/appThunks.ts]: This should be handled properly now. If blind sigs are currently being updated, the app will wait up to 2 minutes for that update to finish. If the update does not finish in time an error will be shown and app linking aborted.

File: BrightID/src/components/Apps/appThunks.ts (hunk 3 above)
Comment (TripleSpeeder) [reply BrightID/src/components/Apps/appThunks.ts]: Fixed!

File: BrightID/src/components/EditProfile/socialMediaThunks.ts (hunk 4 above)
Comment (TripleSpeeder) [reply BrightID/src/components/EditProfile/socialMediaThunks.ts]: Social media registry app linking is now refactored.

**Pull request merged**